### ISSUE TITLE
SUSHI should validate elements of type Code with required bindings

### ISSUE THREAD

**Issue by clarotech** [FHIR/sushi#468]
For example, when defining a CodeSystem you currently can use

* ^hierarchyMeaning = #anything-here

where "anything-here" is not a valid code

**Comment by cmoesel** [comment]
Thanks, @clarotech.  For now, SUSHI does not perform any terminology-related validation.  Terminology will be validated when you run the IG Publisher.

In the future, I think we can make some improvements here -- but we'll still probably be selective about what we validate and how we validate it.  For example, we likely will not want to introduce any dependencies on terminology servers.

**Comment by clarotech** [comment]
I appreciate the boundaries, though for elements of type code, with a required binding - the permissible codes are part of the actual resource definition and as such are fixed per release of FHIR.

For example CodeSystem.status

**Comment by cmoesel** [comment]
Agreed, we could validate any bindings for which the complete code set is available in the core FHIR specification or in VS/CS definitions in dependency IGs.  Those are the types of things we would support; bindings that resolve to externally defined value sets or value sets that are not fully enumerated in their local definition would not be supported.
